### PR TITLE
Add List of nested types support for Arrow vtab: Struct, List, FixedSizeList

### PR DESCRIPTION
**Pull request by sgrebnov**
## 🗣 Description

Add List of nested types (List , Struct, FixedSizeList) support for Arrow vtab.

Record batch from `let rb = stmt.query_arrow(param)?.next().expect("no record batch");` to show the data used in added tests

```yaml
---- vtab::arrow::test::test_list_of_fixed_size_lists_roundtrip stdout ----
+--------------------+
| a                  |
+--------------------+
| [[1, 2], , [3, 4]] |
|                    |
+--------------------+
```

```yaml
---- vtab::arrow::test::test_list_of_structs_roundtrip stdout ----
+----------------------------------+
| a                                |
+----------------------------------+
| [{i: 1, s: foo}, {i: 2, s: baz}] |
|                                  |
| [{i: 5, s: baz}]                 |
+----------------------------------+
```

```yaml
---- vtab::arrow::test::test_list_of_lists_roundtrip stdout ----
+----------------------+
| a                    |
+----------------------+
| [[1, 2], , [3, , 5]] |
|                      |
+----------------------+
```

### CODE REVIEW

File: crates/duckdb/src/vtab/arrow.rs [hunk 1]
```diff
@@ -1442,4 +1450,64 @@ mod test {
...
+
+        check_generic_array_roundtrip(ListArray::new(
+            Arc::new(Field::new(
+                "item",
+                DataType::Struct(vec![field_i, field_s].into()),
+                true,
+            )),
+            OffsetBuffer::new(ScalarBuffer::from(vec![0, 2, 4, 5])),
+            Arc::new(struct_array),
+            Some(vec![true, false, true].into()),
```
Comment (sgrebnov) [inline crates/duckdb/src/vtab/arrow.rs:1508]: This also tests NULL values support

File: crates/duckdb/src/vtab/arrow.rs [hunk 2]
```diff
@@ -592,11 +594,24 @@ fn list_array_to_vector<O: OffsetSizeTrait + AsPrimitive<usize>>(
...
+        DataType::FixedSizeList(_, _) => {
+            fixed_size_list_array_to_vector(as_fixed_size_list_array(value_array.as_ref()), &mut out.array_child())?;
+        }
+        DataType::Struct(_) => {
+            struct_array_to_vector(as_struct_array(value_array.as_ref()), &mut out.struct_child())?;
+        }
         _ => {
-            return Err("Nested list is not supported yet.".into());
+            return Err(format!(
+                "List with elements of type '{}' is not currently supported.",
```
Comment (phillipleblanc) [inline crates/duckdb/src/vtab/arrow.rs:611]:
```suggestion
                "List with elements of type '{}' are not currently supported.",
```

**Review by phillipleblanc — APPROVED** [review]
